regolith/builders/preslistbuilder.py

```diff
@@ -31,30 +31,81 @@ def construct_global_ctx(self):
         gtx['str'] = str
         gtx['zip'] = zip
 
+    def group_member_ids(self, grp):
+        """Get a list of all group member ids
+
+        Parameters
+        ----------
+        grp: string
+            The id of the group
+
+        Returns
+        -------
+        list:
+            The list of ids of the people in the group
+
+        Notes
+        -----
+        - Groups that are being tracked are listed in the groups collection
+        with a name and an id.
+        - People are in a group during an educational or employment period.
+        - To assign a person to a tracked group during one such period, add
+        a "group" key to that education/employment item with a value
+        that is the group name.
+        - This function takes the group id that is passed and searches
+        the people collection for all people that have been
+        assigned to that group in some period of time and returns a list of
+        """
+        grpmembers = []
+        for person in self.gtx['people']:
+            for position in person.get('education', {}):
+                if position.get('group', None) == grp:
+                    if person['_id'] not in grpmembers:
+                        grpmembers.append(person['_id'])
+            for position in person.get('employment', {}):
+                if position.get('group', None) == grp:
+                    if person['_id'] not in grpmembers:
+                        grpmembers.append(person['_id'])
+        return grpmembers
+
     def latex(self):
         """Render latex template"""
         for group in self.gtx['groups']:
-            pi = fuzzy_retrieval(self.gtx['people'], ['aka', 'name', '_id'],
-                                 group['pi_name'])
-
-        presentationsdict = deepcopy(self.gtx['presentations'])
-        for pres in presentationsdict:
-            pauthors = pres['authors']
-            if isinstance(pauthors, str):
-                pauthors = [pauthors]
-            pres['authors'] = [
-                fuzzy_retrieval(self.gtx['people'], ['aka', 'name', '_id'],
-                                author)['name'] for author in pauthors]
-
-            authorlist = ', '.join(pres['authors'])
-            pres['authors'] = authorlist
-            if 'institution' in pres:
-                pres['institution'] = fuzzy_retrieval(self.gtx['institutions'],
-                                                      ['aka', 'name', '_id'],
-                                                      pres['institution'])
-                if 'department' in pres:
-                    pres['department'] = pres['institution']['departments'][
-                        pres['department']]
-        self.render('preslist.tex', 'presentations.tex', pi=pi,
-                    presentations=presentationsdict)
-        self.pdf('presentations')
+            grp = group['_id']
+            grpmember_ids = self.group_member_ids(grp)
+            for member in grpmember_ids:
```

Comment:
Would this be easier as:
```python
for p in self.gtx['people']:
    if p['_id'] not in grpmember_ids:
        continue
```
(This would skip the rest of the loop for any person not in the group)

This would give you direct access to the person in question.
This might also let you use [`filter_publications`](https://github.com/regro/regolith/blob/master/regolith/tools.py#L105) to filter the presentations by the person.

(I'm drawing inspiration from the [cvbuilder ](https://github.com/regro/regolith/blob/master/regolith/builders/cvbuilder.py))

Reply:
I tried really hard to make something like this work, but the skipped entries were still in the collection, just in the wrong format, so it was breaking the jinja2 when it was passed.  I can go look at ``cvbuilder`` though, as this same problem was presumably solved over there.